devtools/gradle/src/main/java/io/quarkus/gradle/extension/QuarkusPluginExtension.java

```diff
@@ -193,8 +194,11 @@ public SourceSetExtension sourceSetExtension() {
         return getSourceSets().getByName(SourceSet.MAIN_SOURCE_SET_NAME).getResources().getSrcDirs();
     }
 
-    public Set<File> outputSourcesDir() {
-        return getSourceSets().getByName(SourceSet.MAIN_SOURCE_SET_NAME).getOutput().getClassesDirs().getFiles();
+    public Set<File> combinedOutputSourceDirs() {
+        Set<File> sourcesDirs = new HashSet<>();
```

Comment:
Do order matter here?

Reply:
I don't think so, but we can definitely add the test directories after main source sets.